### PR TITLE
Adopt Spotless for coding style

### PR DESCRIPTION
**Pull request by peternied**
### Description
This change adds spotless to gradle build process and enforces it in the CI workflow.  There is also a formatter that is used by spotless to apply the rules that is usable via IDEs as well.

### Check List
- [X] New functionality includes testing
  - [X] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java
Comment (peternied) [inline DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java:1]: Note; I've added several files such as this one that help highlight the style differences.  Before merging I'll update the entire codebase, but in its current state is much easier to review

File: build.gradle [hunk 1]
```diff
@@ -1,16 +1,57 @@
...
+    }
+}
+
 subprojects {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
+    apply plugin: "com.diffplug.spotless"
+
+    spotless {
+        java {
```
Comment (peternied) [inline build.gradle:40]: There is a lot of flexibility with the java styling, we can update options here and in the `formatter.xml` to tweak any settings that are must haves.

File: formatter.xml [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_after_and_in_type_parameter" value="insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_after_comma_in_for_inits" value="insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.indent_statements_compare_to_block" value="true"/>
+        <setting id="org.eclipse.jdt.core.formatter.brace_position_for_anonymous_type_declaration" value="end_of_line"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_question_in_wildcard" value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_opening_paren_in_annotation" value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_comma_in_method_invocation_arguments" value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_opening_brace_in_switch" value="insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.comment.align_tags_descriptions_grouped" value="true"/>
+        <setting id="org.eclipse.jdt.core.formatter.comment.line_length" value="140"/>
```
Comment (chelma) [inline formatter.xml:82]: line length gooooooooood!

**Review by chelma — COMMENTED** [review]
What's the plan for reformatting the codebase to the new standard?

**Comment by peternied** [comment]
> What's the plan for reformatting the codebase to the new standard?

@chelma If we are happy, I'll rebase this PR against main and then run `./gradlew spotlessApply` and it will make all the java code conform to the new format.  If anyone has a change in flight, they can merge just the gradle files, then then run `./gradlew spotlessApply` so they don't have to go through a merge conflict nightmare

**Review by chelma — APPROVED** [review]
> > What's the plan for reformatting the codebase to the new standard?
> 
> @chelma If we are happy, I'll rebase this PR against main and then run `./gradlew spotlessApply` and it will make all the java code conform to the new format. If anyone has a change in flight, they can merge just the gradle files, then then run `./gradlew spotlessApply` so they don't have to go through a merge conflict nightmare

Sounds good to me - thanks!

**Review by AndreKurait — APPROVED** [review]

File: formatter.xml (hunk 2 above)
Comment (peternied) [reply formatter.xml]: Done!  BTW see what this did to the codebase in https://github.com/opensearch-project/opensearch-migrations/pull/784/commits/052481fd42a9895a736b42d7867255c650f5bef4

**Pull request merged**